Automattic/wp-calypso
# Remove deprecated features code

### Changes proposed in this Pull Request

This PR removes legacy code that's related to products features.

### Implementation notes

- Deleted `FEATURE_CATEGORIES`, which is not used anymore, and associated logic
- Removed unused features in `features-list.js` and `features.ts` (I double checked there was no remaining instances in the codebase)
- Deleted the following unused functions: `getValidFeatureKeys`, `getFeatureCategoryByKey`, and `getFeatureTitle `
- Deleted the `MORE_FEATURES_LINK` constant and associated logic
- Deleted the following types: `JetpackPlanCardFeature`, and `JetpackPlanCardFeatureSection`

### Testing instructions

- Download the PR and run Calypso
- Open the browser console
- Visit the plans page for both a simple and Jetpack site
- Make sure it looks like production
- Check that there's no relevant error in the console
